src/ArrowFunctions.cpp

```diff
@@ -273,7 +273,7 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
         auto chpl_ptr = (int64_t*)chpl_arr;
         parquet::Int64Reader* reader =
           static_cast<parquet::Int64Reader*>(column_reader.get());
-        reader->Skip(startIdx);
+        startIdx -= reader->Skip(startIdx);
```

Comment:
I think that I am missing something with these subtractions, is there a reason for subtracting the result of the `Skip` call from `startIdx`? It doesn't look like the value is used again after this and I think that the `Skip()` function will loop until it has skipped the correct number of values.

Reply:
Ooh, my mistake, it is in that loop, of course...